File: p3_c/samples/TSel.c
/** Test Selections:       if, if-else, nested if-else
 *  Logical Operators:     &&, ||, !
 *  Relational Operators:  <, >, ==, <=, >=, !=
 *  Program-ID:     TSel.c
 *  Author:         Tyler Oalman
 *  OS:             Arch Linux
 *  Compiler:       GNU C
 */

#include <stdio.h>

void main ()
{
	int i1=1, i2=2, i3=3, i4=4, i5=5, i6=6;

	/* Test a simple if */
	if (i4 > i1) { printf("i4 > i1\n"); }

	/* Test if-else */
	if ((i5 < i2) && (i3 >= i2)) { printf("(i5 < i2) && (i3 >= i2(\n"); }
	else { printf("(i5 >= i2) || (i3 < i2)\n"); }

	/* Test nested if-else */
	if (i1 != i2) { printf("(i1 != i2)\n"); }
	else
	{
		if ((i4 == i5) || (i5 != i6))
		{
			printf("(i1 == i2) && ((i4 == i5) || (i5 != i6))");
		}
	}
}

File: p6_python/samples/TSel.py
"""
Test Selections:    `if`, `if-else`, nested `if-else`
Logical Operators:  `and`, `or`, `not` (invalid: `&&`, `||`, `!`)
Relational Operators:   `<`, `>`, `==`, `<=`, `>=`, `!=`
Program-ID:         TSel.py
Author:             Tyler Oalman
OS:                 Arch Linux
Interpreter:        Python 3
"""


I1 = 1
I2 = 2
I3 = 3
I4 = 4
I5 = 5
I6 = 6

# Test a simple `if`
if I4 > I1:
    print("I4 > I1")

# Test `if-else`
if (I5 < I2) and (I3 >= I2):
    print("(I5 < I2) and (I3 >= I2)")
else:
    print("(I5 >= I2) and (I3 < I2)")

# Test nested `if-else`
if I1 != I2:
    print("(I1 != I2)")
elif (I4 == I5) or (I5 != I6):
    print("(I1 == I2) and (I4 == i5) or (I5 != I6)")

''' Output
        I4 > I1
        (I5 >= I2) or (I3 < I2)
        (I1 != I2)
'''
